gpMgmt/bin/pythonSrc/PyGreSQL-4.0/pgmodule.c

```diff
@@ -325,6 +386,166 @@ print_result(FILE *fout, const PGresult *res)
 	}
 }
 
+/* format result (mostly useful for debugging) */
+/* Note: This is similar to the Postgres function PQprint().
+ * PQprint() is not used because handing over a stream from Python to
+ * Postgres can be problematic if they use different libs for streams
+ * and because using PQprint() and tp_print is not recommended any more.
+ */
+static PyObject *
+format_result(const PGresult *res)
```

Comment:
Why not modify the existing `print_result` function so that it can be used with both `reprfunc` and `printfunc`?